src/main.c

```diff
@@ -143,36 +145,105 @@ dump_conntrack(GHashTable *ips_to_migrate)
- * 2. Conntrack events threads are started to filter events for the IP
- *    addresses present in the ips_to_migrate. If any update is received for a
- *    non-exisitent entry, it is treated as NEW since it contains the base five
- *    tuple information required to identify flow. Similarly, if a destroy
- *    event is received for a non-existent entry, it is ignored.
- * 3. Finally, Conntrack dump is taken from the kernel to get all the live
- *    flows in the system.
- *
- * The above workflow is used to maintain a local copy of the CT entries for
- * the VM.
- *
- * Args:
- *   @ips_to_migrate Hashtable of IP addresses for which CT entries
- *                   have to be migrated.
- *   @stop_flag flag used by thread to exit after dbus operations.
- *
- * Returns:
- *   0 in case of success, -1 otherwise
- */
+* Start threads required for save mode of operation when filter type is
+* FILTER_BY_CT_ZONE.
+*
+* Following things are performed in the save mode:
+* 1. Conntrack events threads are started to filter events for the CT
+*    zones present in the ct_zones_to_migrate. If any update is received for a
+*    non-exisitent entry, it is treated as NEW since it contains the base five
+*    tuple information required to identify flow. Similarly, if a destroy
+*    event is received for a non-existent entry, it is ignored.
+* 2. Finally, Conntrack dump is taken from the kernel to get all the live
+*    flows in the system.
+*
+* The above workflow is used to maintain a local copy of the CT entries for
+* the VM.
+*
+* Args:
+*   @ct_zones_to_migrate Hashtable of CT zones for which CT entries
+*                   have to be migrated.
+*   @stop_flag flag used by thread to exit after dbus operations.
+*
+* Returns:
+*   0 in case of success, -1 otherwise
+*/
 static int
-start_in_save_mode(GHashTable *ips_to_migrate, bool *stop_flag)
+start_in_save_mode_for_ct_zones(GHashTable *ct_zones_to_migrate,
+                                bool *stop_flag)
+{
+    int ret;
+    uint32_t num_ct_zones;
+    struct ct_events_targs *targs;
+
+    num_ct_zones = g_hash_table_size(ct_zones_to_migrate);
+
+    if (num_ct_zones == 0) {
+        LOG(INFO, "%s: Not starting any save mode threads since number of"
+            "ct-zones is 0", __func__);
+        return 0;
+    }
+
+    targs = g_malloc0(sizeof(struct ct_events_targs));
+    targs->ct_zones_to_migrate = ct_zones_to_migrate;
+    targs->stop_flag = stop_flag;
+    targs->is_src = true;
+    ret = pthread_create(&targs->tid, NULL, &pthread_wrapper_ct_events,
+                         (void *)targs);
+    if (ret != 0) {
+        LOG(ERROR, "%s: src events thread creation failed. %s", __func__,
+            strerror(ret));
+        return -1;
+    }
+    ret = pthread_setname_np(targs->tid, "ct_events_src");
+    if (ret != 0) {
+        LOG(WARNING, "%s: Failed to set thread name \"ct_events_src\". %s",
+            __func__, strerror(ret));
+    }
+
+    dump_conntrack(targs);
```

Comment:
Is there a reason to use event args here? those were meant for different threads.

you can simply pass ct_zones_to_migrate as second arg right?

for IP case you can set it to NULL/empty:

dump_conntrack(ips_to_migrate, NULL);

for the ct-zone case set, ips_to_migrate as NULL/empty.

dump_conntrack(NULL, ct_zones_to_migrate);